### PR TITLE
Misc light platform cleanup

### PR DESCRIPTION
**Pull request by oxan**
# What does this implement/fix? 

Some clean-up I noticed during discussions on #1895, and a missed log message.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes N/A

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** N/A

## Test Environment

- [ ] ESP32
- [x] ESP8266

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/light/light_state.cpp [hunk 1]
```diff
@@ -170,7 +170,7 @@ void LightState::current_values_as_brightness(float *brightness) {
 }
 void LightState::current_values_as_rgb(float *red, float *green, float *blue, bool color_interlock) {
```
Comment (jesserockz) [inline esphome/components/light/light_state.cpp]: Not sure why the `color_interlock` is even a parameter here, maybe remove it?

Comment (jesserockz) [reply esphome/components/light/light_state.cpp]: Same for the others below

Comment (oxan) [reply esphome/components/light/light_state.cpp]: Leftover from when color_interlock wasn't in the traits, I guess. Added a commit to remove them.

**Comment by oxan** [comment]
Given that this one conflicts with the color mode PR, I'll close this one for now and land the contents through new PRs.

**Pull request closed without merging**